File: src/gui/objects/window/stage/WindowState.java
package gui.objects.window.stage;

import gui.objects.window.stage.WindowBase.Maximized;
import layout.container.layout.Layout;
import static main.App.APP;

class WindowState {
	public final double x, y, w, h;
	public final boolean resizable, minimized, fullscreen, onTop;
	public final Maximized maximized;
	public final Layout layout;

	public WindowState(Window window) {
		x = window.X.getValue();
		y = window.Y.getValue();
		w = window.W.getValue();
		h = window.H.getValue();
		resizable = window.resizable.getValue();
		minimized = window.s.iconifiedProperty().getValue();
		fullscreen = window.fullscreen.getValue();
		onTop = window.alwaysOnTop.getValue();
		maximized = window.maximized.getValue();
		layout = window.getLayout();
	}

	public Window toWindow() {
		Window window = APP.windowManager.create(WindowManager.canBeMainTemp);
		window.X.set(x);
		window.Y.set(y);
		window.W.set(w);
		window.H.set(h);
		window.s.setIconified(minimized);
		window.MaxProp.set(maximized);
		window.FullProp.set(fullscreen);
		window.resizable.set(resizable);
		window.setAlwaysOnTop(onTop);
		window.initLayout(layout);
		return window;
	}
}

File: src/sp/it/pl/gui/objects/window/stage/WindowState.kt
package sp.it.pl.gui.objects.window.stage

import sp.it.pl.main.APP

internal class WindowState(window: Window) {
    val x = window.X.value!!
    val y = window.Y.value!!
    val w = window.W.value!!
    val h = window.H.value!!
    val resizable = window.resizable.value!!
    val minimized = window.s.iconifiedProperty().value!!
    val fullscreen = window.fullscreen.value!!
    val onTop = window.alwaysOnTop.value!!
    val maximized = window.maximized.value!!
    val layout = window.layout!!

    fun toWindow(): Window {
        val window = APP.windowManager.create(WindowManager.canBeMainTemp)
        window.X.set(x)
        window.Y.set(y)
        window.W.set(w)
        window.H.set(h)
        window.s.isIconified = minimized
        window.MaxProp.set(maximized)
        window.FullProp.set(fullscreen)
        window.resizable.set(resizable)
        window.isAlwaysOnTop = onTop
        window.initLayout(layout)
        return window
    }
}